### PR TITLE
python3Packages.uvloop: 0.14.0 -> 0.15.0

### PR DESCRIPTION
**Pull request by mweinelt**
###### Motivation for this change

https://github.com/MagicStack/uvloop/releases/tag/v0.15.0

Reenable all tests, drop darwin sandbox patch, cleanup. This unbreaks uvloop on python3.9.

Would like someone on x86_64-darwin to test this, especially with regard to the testsuite.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by mweinelt** [comment]
Result of `nixpkgs-review pr 112657` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.starlette</li>
    <li>python39Packages.starlette</li>
  </ul>
</details>
<details>
  <summary>63 packages built:</summary>
  <ul>
    <li>aiodnsbrute</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>google-music-scripts</li>
    <li>home-assistant</li>
    <li>mtprotoproxy</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.aioextensions</li>
    <li>python37Packages.aiorun</li>
    <li>python37Packages.anyio</li>
    <li>python37Packages.asyncpg</li>
    <li>python37Packages.axis</li>
    <li>python37Packages.databases</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.ircrobots</li>
    <li>python37Packages.orm</li>
    <li>python37Packages.pproxy</li>
    <li>python37Packages.qiskit</li>
    <li>python37Packages.qiskit-ibmq-provider</li>
    <li>python37Packages.respx</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python37Packages.starlette</li>
    <li>python37Packages.uvicorn</li>
    <li>python37Packages.uvloop</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aioextensions</li>
    <li>python38Packages.aiorun</li>
    <li>python38Packages.anyio</li>
    <li>python38Packages.asyncpg</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.databases</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.ircrobots</li>
    <li>python38Packages.orm</li>
    <li>python38Packages.pproxy</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-ibmq-provider</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.uvloop</li>
    <li>python39Packages.aioextensions</li>
    <li>python39Packages.aiorun</li>
    <li>python39Packages.asyncpg</li>
    <li>python39Packages.databases</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.orm</li>
    <li>python39Packages.pproxy</li>
    <li>python39Packages.uvicorn</li>
    <li>python39Packages.uvloop</li>
    <li>sourcehut.listssrht</li>
    <li>theharvester</li>
  </ul>
</details>

The starlette breakage will be fixed in #111079.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 112657` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>home-assistant</li>
    <li>python38Packages.fastapi</li>
    <li>python38Packages.starlette</li>
    <li>python39Packages.starlette</li>
  </ul>
</details>
<details>
  <summary>62 packages built:</summary>
  <ul>
    <li>aiodnsbrute</li>
    <li>datasette (python38Packages.datasette)</li>
    <li>google-music-scripts</li>
    <li>mtprotoproxy</li>
    <li>python37Packages.PyRMVtransport</li>
    <li>python37Packages.aioextensions</li>
    <li>python37Packages.aiorun</li>
    <li>python37Packages.anyio</li>
    <li>python37Packages.asyncpg</li>
    <li>python37Packages.axis</li>
    <li>python37Packages.databases</li>
    <li>python37Packages.datasette</li>
    <li>python37Packages.entrance</li>
    <li>python37Packages.entrance-with-router-features</li>
    <li>python37Packages.fastapi</li>
    <li>python37Packages.google-music</li>
    <li>python37Packages.httpcore</li>
    <li>python37Packages.httpx</li>
    <li>python37Packages.ircrobots</li>
    <li>python37Packages.orm</li>
    <li>python37Packages.pproxy</li>
    <li>python37Packages.qiskit</li>
    <li>python37Packages.qiskit-ibmq-provider</li>
    <li>python37Packages.respx</li>
    <li>python37Packages.sanic</li>
    <li>python37Packages.sanic-auth</li>
    <li>python37Packages.starlette</li>
    <li>python37Packages.uvicorn</li>
    <li>python37Packages.uvloop</li>
    <li>python38Packages.PyRMVtransport</li>
    <li>python38Packages.aioextensions</li>
    <li>python38Packages.aiorun</li>
    <li>python38Packages.anyio</li>
    <li>python38Packages.asyncpg</li>
    <li>python38Packages.axis</li>
    <li>python38Packages.databases</li>
    <li>python38Packages.entrance</li>
    <li>python38Packages.entrance-with-router-features</li>
    <li>python38Packages.google-music</li>
    <li>python38Packages.httpcore</li>
    <li>python38Packages.httpx</li>
    <li>python38Packages.ircrobots</li>
    <li>python38Packages.orm</li>
    <li>python38Packages.pproxy</li>
    <li>python38Packages.qiskit</li>
    <li>python38Packages.qiskit-ibmq-provider</li>
    <li>python38Packages.respx</li>
    <li>python38Packages.sanic</li>
    <li>python38Packages.sanic-auth</li>
    <li>python38Packages.uvicorn</li>
    <li>python38Packages.uvloop</li>
    <li>python39Packages.aioextensions</li>
    <li>python39Packages.aiorun</li>
    <li>python39Packages.asyncpg</li>
    <li>python39Packages.databases</li>
    <li>python39Packages.httpcore</li>
    <li>python39Packages.orm</li>
    <li>python39Packages.pproxy</li>
    <li>python39Packages.uvicorn</li>
    <li>python39Packages.uvloop</li>
    <li>sourcehut.listssrht</li>
    <li>theharvester</li>
  </ul>
</details>

**Comment by mweinelt** [comment]
> home-assistant

Yeah, I would really like more logs from you on that badboy.

**Review by fabaff — COMMENTED** [review]
- [x] package build (see https://github.com/NixOS/nixpkgs/pull/112657#issuecomment-776778255)
- [ ] all depending packages build
- [x] tested on x86_64

```bash
...ew/pr-112657/results/python38Packages.uvloop/lib/python3.8/site-packages]$ python
Python 3.8.7 (default, Dec 21 2020, 17:18:55) 
[GCC 10.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import uvloop
>>> import asyncio
>>> asyncio.set_event_loop_policy(uvloop.EventLoopPolicy())
>>> async def main():
...     print(1)
... 
>>> uvloop.install()
>>> asyncio.run(main())
1
```

Sees to work :smile:

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112657` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>

  <summary>3 packages failed to build and are new build failures:</summary>
  <ul>
<li>python38Packages.fastapi: log was empty</li>
<li>python38Packages.starlette: log <a href="https://termbin.com/nvv3">https://termbin.com/nvv3</a></li>
<li>python39Packages.starlette: log <a href="https://termbin.com/0io8">https://termbin.com/0io8</a></li>
  </ul>
</details>
<details>
<summary>
63 packages built:
</summary>
<ul>
<li>aiodnsbrute</li>
<li>datasette (python38Packages.datasette)</li>
<li>google-music-scripts</li>
<li>home-assistant</li>
<li>mtprotoproxy</li>
<li>python37Packages.PyRMVtransport</li>
<li>python37Packages.aioextensions</li>
<li>python37Packages.aiorun</li>
<li>python37Packages.anyio</li>
<li>python37Packages.asyncpg</li>
<li>python37Packages.axis</li>
<li>python37Packages.databases</li>
<li>python37Packages.datasette</li>
<li>python37Packages.entrance</li>
<li>python37Packages.entrance-with-router-features</li>
<li>python37Packages.fastapi</li>
<li>python37Packages.google-music</li>
<li>python37Packages.httpcore</li>
<li>python37Packages.httpx</li>
<li>python37Packages.ircrobots</li>
<li>python37Packages.orm</li>
<li>python37Packages.pproxy</li>
<li>python37Packages.qiskit</li>
<li>python37Packages.qiskit-ibmq-provider</li>
<li>python37Packages.respx</li>
<li>python37Packages.sanic</li>
<li>python37Packages.sanic-auth</li>
<li>python37Packages.starlette</li>
<li>python37Packages.uvicorn</li>
<li>python37Packages.uvloop</li>
<li>python38Packages.PyRMVtransport</li>
<li>python38Packages.aioextensions</li>
<li>python38Packages.aiorun</li>
<li>python38Packages.anyio</li>
<li>python38Packages.asyncpg</li>
<li>python38Packages.axis</li>
<li>python38Packages.databases</li>
<li>python38Packages.entrance</li>
<li>python38Packages.entrance-with-router-features</li>
<li>python38Packages.google-music</li>
<li>python38Packages.httpcore</li>
<li>python38Packages.httpx</li>
<li>python38Packages.ircrobots</li>
<li>python38Packages.orm</li>
<li>python38Packages.pproxy</li>
<li>python38Packages.qiskit</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.respx</li>
<li>python38Packages.sanic</li>
<li>python38Packages.sanic-auth</li>
<li>python38Packages.uvicorn</li>
<li>python38Packages.uvloop</li>
<li>python39Packages.aioextensions</li>
<li>python39Packages.aiorun</li>
<li>python39Packages.asyncpg</li>
<li>python39Packages.databases</li>
<li>python39Packages.httpcore</li>
<li>python39Packages.orm</li>
<li>python39Packages.pproxy</li>
<li>python39Packages.uvicorn</li>
<li>python39Packages.uvloop</li>
<li>sourcehut.listssrht</li>
<li>theharvester</li>
</ul>
</details>

**Comment by fabaff** [comment]
Comments in #111079 addressed.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 112657` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.fastapi</li>
<li>python37Packages.starlette</li>
<li>python38Packages.fastapi</li>
<li>python38Packages.starlette</li>
<li>python39Packages.fastapi</li>
<li>python39Packages.google-music</li>
<li>python39Packages.starlette</li>
</ul>
</details>
<details>

  <summary>16 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.aioextensions: log <a href="https://termbin.com/pzc8">https://termbin.com/pzc8</a></li>
<li>python37Packages.entrance: log was empty</li>
<li>python37Packages.entrance-with-router-features: log was empty</li>
<li>python37Packages.qiskit: log was empty</li>
<li>python37Packages.sanic: log <a href="https://termbin.com/qggc">https://termbin.com/qggc</a></li>
<li>python37Packages.sanic-auth: log was empty</li>
<li>python38Packages.aioextensions: log <a href="https://termbin.com/q2s8">https://termbin.com/q2s8</a></li>
<li>python38Packages.entrance: log was empty</li>
<li>python38Packages.entrance-with-router-features: log was empty</li>
<li>python38Packages.qiskit: log was empty</li>
<li>python38Packages.sanic: log <a href="https://termbin.com/0r4ef">https://termbin.com/0r4ef</a></li>
<li>python38Packages.sanic-auth: log was empty</li>
<li>python39Packages.entrance: log was empty</li>
<li>python39Packages.entrance-with-router-features: log was empty</li>
<li>python39Packages.sanic: log <a href="https://termbin.com/qwx2w">https://termbin.com/qwx2w</a></li>
<li>python39Packages.sanic-auth: log was empty</li>
  </ul>
</details>
<details>
<summary>
56 packages built:
</summary>
<ul>
<li>aiodnsbrute</li>
<li>datasette (python38Packages.datasette)</li>
<li>google-music-scripts</li>
<li>home-assistant</li>
<li>mtprotoproxy</li>
<li>python37Packages.PyRMVtransport</li>
<li>python37Packages.aiorun</li>
<li>python37Packages.anyio</li>
<li>python37Packages.asyncpg</li>
<li>python37Packages.axis</li>
<li>python37Packages.databases</li>
<li>python37Packages.datasette</li>
<li>python37Packages.google-music</li>
<li>python37Packages.httpcore</li>
<li>python37Packages.httpx</li>
<li>python37Packages.ircrobots</li>
<li>python37Packages.orm</li>
<li>python37Packages.pproxy</li>
<li>python37Packages.qiskit-ibmq-provider</li>
<li>python37Packages.respx</li>
<li>python37Packages.uvicorn</li>
<li>python37Packages.uvloop</li>
<li>python38Packages.PyRMVtransport</li>
<li>python38Packages.aiorun</li>
<li>python38Packages.anyio</li>
<li>python38Packages.asyncpg</li>
<li>python38Packages.axis</li>
<li>python38Packages.databases</li>
<li>python38Packages.google-music</li>
<li>python38Packages.httpcore</li>
<li>python38Packages.httpx</li>
<li>python38Packages.ircrobots</li>
<li>python38Packages.orm</li>
<li>python38Packages.pproxy</li>
<li>python38Packages.qiskit-ibmq-provider</li>
<li>python38Packages.respx</li>
<li>python38Packages.uvicorn</li>
<li>python38Packages.uvloop</li>
<li>python39Packages.PyRMVtransport</li>
<li>python39Packages.aioextensions</li>
<li>python39Packages.aiorun</li>
<li>python39Packages.anyio</li>
<li>python39Packages.asyncpg</li>
<li>python39Packages.axis</li>
<li>python39Packages.databases</li>
<li>python39Packages.datasette</li>
<li>python39Packages.httpcore</li>
<li>python39Packages.httpx</li>
<li>python39Packages.ircrobots</li>
<li>python39Packages.orm</li>
<li>python39Packages.pproxy</li>
<li>python39Packages.respx</li>
<li>python39Packages.uvicorn</li>
<li>python39Packages.uvloop</li>
<li>sourcehut.listssrht</li>
<li>theharvester</li>
</ul>
</details>

**Comment by mweinelt** [comment]
> 16 packages failed to build and are new build failures:

Can you please check these aren't broken on master before claiming they are *new*? None of these looks related to uvloop. @SuperSandro2000

**Comment by mweinelt** [comment]
Please don't tell me these are new build failures if they clearly aren't. I took the liberty to check hydra and just a few examples (on python38) those have been failing quite consistently on master.

- https://hydra.nixos.org/job/nixpkgs/trunk/python38Packages.qiskit.x86_64-darwin
- https://hydra.nixos.org/job/nixpkgs/trunk/python38Packages.entrance.x86_64-darwin
- https://hydra.nixos.org/job/nixpkgs/trunk/python38Packages.entrance-with-router-features.x86_64-darwin
- https://hydra.nixos.org/job/nixpkgs/trunk/python38Packages.sanic.x86_64-darwin
- https://hydra.nixos.org/job/nixpkgs/trunk/python38Packages.sanic-auth.x86_64-darwin

**Comment by SuperSandro2000** [comment]
> Can you please check these aren't broken on master before claiming they are _new_?

I am not checking them by hand. The automated check was broken for quite some while but I thought I fixed them. Maybe I didn't deploy it yet to darwin. Investigating.

**Comment by mweinelt** [comment]
Since the uvloop package builds on x86_64-darwin and the build failures seem mostly unrelated I think this is fine.
